Makefile.am

```diff
@@ -163,7 +164,18 @@ clean-storage:
 all-local: \
 	vdsm.spec
 
-.PHONY: srpm rpm
+.PHONY: config
+config:
+	./autogen.sh --system --enable-timestamp
+
+.PHONY: dist-build
+dist-build: | clean config all clean-rpm rpm
```

Comment:
We cannot use config here, but otherwise using make targets for building and installing looks good.

However if we need to run autogen.sh in some cases, for example for after adding missing tags, it is better to add external script for deploying, for example:

    # dev-install.sh
    #!/bin/sh
    The long one liner to clean, build, and upgrade the rpms...

Reply:
I wonder, actually reading this, if it would work if instead of having the targets as dependencies we do:

```
dist-build: | clean config
    make all && make clean-rpm && make rpm
```

In theory these are new instances of the newly generate Makefile, so it may work. Anyway, it is obscure and dangerous, so I will just remove the `config` target as the proper solution.